third_party/amd/backend/compiler.py

```diff
@@ -220,6 +220,8 @@ def make_amdgcn(src, metadata, options):
         # llvm -> hsaco
         amdgcn = llvm.translate_to_asm(src, amd.TARGET_TRIPLE, options.arch, '', [], options.enable_fp_fusion, False)
         if os.environ.get("AMDGCN_ENABLE_DUMP", "0") == "1":
+            os.environ["ASM_ENABLE_DUMP"] = "1"
+        if os.environ.get("ASM_ENABLE_DUMP", "0") == "1":
```

Comment:
Do you still need to print here as I thought this might print the asm a second time?

Reply:
Ah yeah, good point I see how this works a little better now. So what I am also wondering is, should we print from the Python or from the C++ side?

Take a look at: https://github.com/openai/triton/blob/main/third_party/nvidia/backend/compiler.py#L237-L253

The python does some subtle modification to the PTX before handing it over to PTXAS. Should we have two different env vars? One for what comes directly out of LLVM and one for what Triton produces?

Reply:
I am considering moving this entirely to 

third_party/nvidia/backend/compiler.py and third_party/amd/backend/compiler.py because it seems like the true PTX dumped out is

Reply:
What do you think about adding a `NVPTX_ENABLE_DUMP=1` in third_party/nvidia/backend/compiler.py to dump the textually modified PTX by Triton and leave  ASM_ENABLE_DUMP to dump the output straight out of LLVM? At the same time, I think we can leave AMDGCN_ENABLE_DUMP alone outside of removing it from the GetEnv table, as it is not needed there.

Reply:
I see. Sounds good to dump the final PTX instead of the pre-modified one. A unified switch woulds still be cleaner. Let's see Lixun's thought.

Reply:
IIUC, the `ASM_ENABLE_DUMP` is trying to dump whatever is the output of LLVM (can be modified a bit)? 
If so, on NV it's dumping ptx and on AMD it's dumping amdgcn. Though they are both output of LLVM, they are at different levels from the machine code perspective. I think amdgcn can be called as assembly. But ptx is "higher" than that. 
We don't have downstream tools relying on the name of the env var (or can be trivially adjusted). But sharing the same env var between amdgcn and ptx is just confusing.